source-pendo/source_pendo/resources.py

```diff
@@ -0,0 +1,181 @@
+import calendar
+from datetime import timedelta
+import functools
+from logging import Logger
+from typing import AsyncGenerator
+import pendulum
+
+from estuary_cdk.flow import CaptureBinding
+from estuary_cdk.capture import common, Task
+from estuary_cdk.http import HTTPMixin
+
+
+from .models import (
+    EndpointConfig,
+    ResourceConfig,
+    ResourceState,
+    Resource,
+    Event,
+    Metadata,
+    RESOURCE_TYPES,
+    EVENT_TYPES,
+    METADATA_TYPES,
+)
+from .api import (
+    fetch_resources,
+    fetch_events,
+    fetch_metadata,
+)
+
+def resources(
+        log: Logger, http: HTTPMixin, config: EndpointConfig
+) -> list[common.Resource]:
+
+    async def snapshot(http: HTTPMixin, entity: str, log: Logger) -> AsyncGenerator[Metadata, None]:
```

Comment:
I know this seems like a pattern in other connectors with snapshot bindings, but it seems like we should be able to have `fetch_resources` (etc.) itself be a generator, and then use it directly (with partial function application) as the `fetch_snapshot` parameter. Does that not work for some reason?

Reply:
Making `fetch_resources` be the generator itself does work. I'll update both it and `fetch_metadata` so we can get rid of the `snapshot` functions.